### PR TITLE
K8SPSMDB-607: Fix operator crashing on restore with `backupSource` and Azure Storage

### PR DESCRIPTION
**Pull request by pooknull**
[K8SPSMDB-607](https://jira.percona.com/browse/K8SPSMDB-607) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://jira.percona.com/browse/K8SPSMDB-607

### CODE REVIEW

**Review by defbin — APPROVED** [review]

File: pkg/apis/psmdb/v1/perconaservermongodbrestore_types.go [hunk 1]
```diff
@@ -82,7 +83,7 @@ func (r *PerconaServerMongoDBRestore) CheckFields() error {
 			return errors.New("backupSource destination is required")
 		}
 
-		if r.Spec.BackupSource.Destination[:5] != "s3://" {
+		if r.Spec.BackupSource.S3 != nil && !strings.HasPrefix(r.Spec.BackupSource.Destination, "s3://") {
```
Comment (egegunes) [inline pkg/apis/psmdb/v1/perconaservermongodbrestore_types.go:86]: should we add validation for the azure too?

**Review by egegunes — APPROVED** [review]

File: pkg/apis/psmdb/v1/perconaservermongodbrestore_types.go (hunk 1 above)
Comment (pooknull) [reply pkg/apis/psmdb/v1/perconaservermongodbrestore_types.go:86]: @egegunes I don't think we should. Azure `BackupSource.Destination` doesn't have any prefixes as s3 has

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| limits | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| scheduled-backup | passed |
| arbiter | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| smart-update | passed |
| upgrade | passed |
| version-service | passed |
| upgrade-sharded | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/825/commits/901bdf56d3d49c02efb6058b9845cc0336f33f45
image: `perconalab/percona-server-mongodb-operator:PR-825-901bdf56`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| limits | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| scheduled-backup | passed |
| arbiter | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| upgrade | passed |
| data-sharded | passed |
| upgrade-sharded | passed |
| non-voting | passed |
| pitr | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/825/commits/864e1602359e1f32ad634e0198c3d8d9c7e0259b
image: `perconalab/percona-server-mongodb-operator:PR-825-864e1602`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| storage | passed |
| data-sharded | passed |
| self-healing | passed |
| non-voting | passed |
| self-healing-chaos | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/825/commits/864e1602359e1f32ad634e0198c3d8d9c7e0259b
image: `perconalab/percona-server-mongodb-operator:PR-825-864e1602`

**Pull request merged**